### DISCUSSION TITLE
Get & activate open child windows.

### DISCUSSION THREAD

**Discussion by Quetzalcoatl2435** [AvaloniaUI/Avalonia#11101] [Q&A] [answered]
Hi, I've got an app that uses a main window with menus that act like a parent window and opening child windows from the menu. Is there a way to get an open child window and activate that window?

E.g.:
1. Click on Menu 1 -> opens Window 1.
2. Click on Menu 2 -> opens Window 2,
3. Click on Menu 1 again -> activate already open Window 1 instead of creating a new Window 1.

I can do this in WPF by giving each window a name and use
`foreach (Window openWindow in App.Current.Windows)`
`    if (openWindow.IsVisible && openWindow.Name == name)`
`        openWindow.Focus();`
but haven't found anything similar in Avalonia.

Thanks!

**Comment by maxkatz6** [comment]
Focus and activation are different concepts (probably not on Windows though).
It seems like you need `openWindow.Activate();`

**Reply by Quetzalcoatl2435** [reply]
Yes, that was just an example that works on WPF where I could iterate over all open windows and find the window I'm looking for. If not found, I can then open a new instance of that window.

What I need is a way to do the same with Avalonia. Either iterate over open windows or directly to find the `openWindow`, which I can then activate.

**Reply by maxkatz6** [reply]
(App.Current as IClassicDesktopStyleApplicationLifetime)?.Windows

Keep in mind, windows exist only on desktop OSes, that's why it's not part of crossplatform Application class directly.
And use `.Activate()` on needed window.

**Reply by Quetzalcoatl2435** [reply]
I'm getting `System.InvalidCastException: 'Unable to cast object of type 'AvaloniaTestApp.App' to type 'Avalonia.Controls.ApplicationLifetimes.IClassicDesktopStyleApplicationLifetime'.'`

Started the app with `BuildAvaloniaApp().StartWithClassicDesktopLifetime(args)`.

**Reply by maxkatz6** [reply, marked as answer]
App.Current.ApplicationLifetime

**Reply by Quetzalcoatl2435** [reply]
OK. That works. Thanks a lot!
